### ISSUE TITLE
Not working in MIGx

### ISSUE THREAD

**Issue by tyreal2012** [alanpich/tvImagePlus#15]
When ever you try to allocate a TV as an Image+ in MIGx grid the image tv doesnt appear.

**Comment by alanpich** [comment]
Was this a fresh install or an upgrade from v1.x ?

**Comment by twincreekmedia** [comment]
+1 for getting this to play nice with MIGX (latest version). Running MODX 2.2.7 with fresh install of Image+ and MGIX. Both work without issues when used independently just not when Image+ is added as a field.

Error in console (ext-all.js:7)

Uncaught TypeError: undefined is not a function
create
Ext.extend.load
(anonymous function)
h.Event.fire
v.onDocumentReady
(anonymous function)
(anonymous function)
B
F

![Screen Shot 2013-04-19 at 3 47 39 PM](https://f.cloud.github.com/assets/625649/404337/3352ef5a-a943-11e2-8446-1bd5870aae94.png)

**Comment by MarkOdey** [comment]
I am supporting this fix. It would be amazing if this issue could be fixed.  One of my client is complaining of the fact that she cant control the cropping position of her pictures with phpthumbof.  It would be an amazing to work with migx so I can implement this in her image galleries. 

thanks Alanpich for your work

Mark

**Comment by alanpich** [comment]
OK.. slightly pre-emptive Boom. Initially renders fine (which is more than it did before) but still throws errors because the required config params aren't being loaded. Might have to load the config via ajax instead of smart-templating them onto the page

**Comment by MarkOdey** [comment]
Hey allan, I am currently trying to fix it on my side too. It seems like its a problem from previous version of migx. However, I am unable to upgrade migx from the latest version. I have tried it on a new install and it worked fine.  I just had to create a new image renderer for the grid 

Thats what i did very basic...

$renderer['this.renderImagePlus'] = "
    renderImagePlus : function(val, md, rec, row, col, s){
        var source = s.pathconfigs[col];
        var obj = JSON.parse(val);
        alert(obj.sourceImg.src); 

```
    var val=obj.sourceImg.src;

    if (val !== null) {
        if (val.substr(0,4) == 'http'){
            return '{$httpimg}' ;
        }        
        if (val != ''){
            return '{$phpthumbimg}';
        }
        return val;
    }
}
```

";

I just wish to find out how I could make it work for the previous version.  I am about to go mad

**Comment by MarkOdey** [comment]
Ok little update on the investigation for making this tv work with migx. By accident I have attached the tv on the resource where there is also my migx / gallery grid attached on. By surprise it worked!!! I feel it is a matter of instantiation. maybe the plugin instantiating should trigger a portion of the codes everytime a resource is loaded. 

hope this help. I will work on it further to fix it 

MARk

**Comment by alanpich** [comment]
Yeah, it was a combination of javascripts not being loaded by MIGX (hence why it worked when there was a non-migx instance on the same page) and some of the config properties not being set properly. 

99% certain it now works

**Comment by lithiumlab** [comment]
Can't make it work inside a MIGXdb form. If i change the tv input type to image its works inmediately.
Tried both ways: creating a helper TV and also using the config settings.
I even included additional code in  MYCONFIG.config.inc.php

$this->customconfigs['includeTVs'] = 1;
$this->customconfigs['includeTVList'] = 'migx_test_image';

If you have one working... Can you show me a printscreen of your configuration?
Maybe i'm missing something

**Comment by alanpich** [comment]
Doe anything appear, or just an empty text box? I'd imagine there are probably javascript errors in the console along the lines of 'undefined is not a function' ?

Haven't used migx db myself but if it works like to regular migx grid then this should be solved by v2.2 due very soon.

**Comment by Glidias** [comment]
It's an undefined error with with the ext-js file line. Origin function is anonymous though from the stack trace.  As MarkOdey mentioned earlier, the error can be prevented (and the imagePlus widgets show up on MigX window), so long if you attached a dummy imagePlus TV on the resource because i think it indirectly loads the dependant files beforehand. I traced the method under includeScriptAssets() and it still runs with/without the imagePlus, so maybe something else is missing or something that causes the undefined error.....  

MarkOdey: "By accident I have attached the tv on the resource where there is also my migx / gallery grid attached on. By surprise it worked!"

Actually, it allows the imagePlus widgets to appear, but it doesn't work properly when it comes to saving the values and retrieving them back in the MigX's getImageList snippet , where the dummy TV value is retrieved instead of each MIGx's entry item's value.... I dunno why this is the case, because when I reload back the MigX window in the manager, the values seem to be saved correctly inside based on what I entered in MigX. Only the frontend getImageList output placeholder seems to have problems retrieving back those values, but instead retrieves it from the original dummy TV value instead.

Also for some odd case, the Output value using getImageList seems to be permanently locked and it can't change!!!
